### PR TITLE
Create event for users joining a room in Matrix

### PR DESCRIPTION
**Pull request by awesome-michael**
# Description
Listens to matrix state events and creates a JoinRoom event when a `m.room.member` event with content `"membership": "join"` occurs.

## Status
**READY**

## Type of change
- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Tested with this skill:
```
@match_event(JoinRoom)
    async def hello_join(self, event):
        text = random.choice(
            ["Welcome {}", "Hey {}, nice to meet you"]).format(event.user)
        await event.respond(Message(text))
```

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: opsdroid/events.py [hunk 1]
```diff
@@ -451,6 +451,9 @@ def __init__(self, description, *args, **kwargs):  # noqa: D107
 class JoinRoom(Event):
     """Event class to represent a user joining a room."""
 
+    def __init__(self, *args, **kwargs):  # noqa: D107
```
Comment (Cadair) [inline opsdroid/events.py]: Do we need an `__init__` here if it's not doing anything?

Comment (awesome-michael) [reply opsdroid/events.py]: Was not sure if its need to call the `super.init`. As no additional parameters are needed.. what would be the correct constructor?

**Comment by awesome-michael** [comment]
> Looks good, it would be great to add a test for this, especially as it's a new field in the sync. Do you have the time to do that?

Added a test. Could you please review it?

File: opsdroid/events.py (hunk 1 above)
Comment (jacobtomlinson) [reply opsdroid/events.py]: If you omit the `__init__` it will call the super init so don't worry. You can remove this.

**Comment by Cadair** [comment]
Thanks a lot @awesome-michael :rocket:

**Pull request merged**